Event handler for Cloud (#4034)
centreon/centreon-documentation: cloud/monitoring/event-handler.md

### Enabling Event Handlers

Event handlers can be enabled or disabled on a poller.

Host- and service-specific event handlers can be enabled or disabled in your host and service definitions. Host- and
service-specific event handlers will not be executed if the global option is disabled on your poller.

### Event Handler Execution Order

As already mentioned, global host and service event handlers are executed immediately before host- or service-specific
event handlers.

Event handlers are executed for HARD problem and recovery states immediately after notifications are sent out.

## Configuration

### Step 1: Create a command

Event handler commands will likely be shell or perl scripts, but they can be any type of executable that can run from a
command prompt. As a minimum, the scripts should take the following macros as arguments:

* For Services: $SERVICESTATE$, $SERVICESTATETYPE$, $SERVICEATTEMPT$
* For Hosts: $HOSTSTATE$, $HOSTSTATETYPE$, $HOSTATTEMPT$

The scripts should examine the values of the arguments passed to it and take any necessary action based upon those values.

> Event handler commands will normally be executed with the same permissions as the user under which Centreon Engine
> (the **centreon-engine** user) is running on your machine. This can present a problem if you want to write an event handler that
> restarts system services, as **root** privileges are generally required to perform tasks of that kind.

Reviewer: ```suggestion
Event handler commands are typically executed with the same permissions as the user running Centreon Engine (the `centreon-engine` user) on your machine. This means that the executed file must have the appropriate execution permissions for this user. If your event handler needs to restart system services, be cautious, as such tasks generally require `root` privileges, and improper handling could lead to security risks or execution failures.
```

Ideally you should evaluate the types of event handlers you will be implementing and grant just enough permissions to
the Centreon Engine user to execute the necessary system commands. You might want to try using sudo to accomplish this.

1. To create a command, go to **Configuration > Commands > Miscellaneous**.
2. [Create your command as described here](./basic-objects/commands.md#creating-a-custom-check-command).
3. Make sure to [add your new command to the command whitelist of the poller](./basic-objects/commands.md#command-whitelist) that will monitor the resources you want.

### Step 2: Enable event handlers on your platform

1. Go to **Configuration > Pollers > Engine configuration** and edit the Centreon Engine configurations for all the servers that will monitor resources with event handlers.

2. In the **Check Options** tab, in the **Misc Options** section, make sure **Event Handler Option** is enabled.

### Step 3: Configure event handlers for resources

#### For all resources monitored by a specific poller

To define an event handler command for all hosts and/or services monitored by a specific poller:

> 1. Go to **Configuration > Pollers > Engine configuration** and edit the Centreon Engine configuration for this poller.

Reviewer 2: User cloud don't have access to this page

---
id: event-handler
title: Event handler (auto remediation)
---

## Introduction

Event handlers are optional system commands (scripts or executables) that are run whenever a host
or service state change occurs.

An obvious use for event handlers is the ability for Centreon to proactively fix problems before anyone is notified.

### When Are Event Handlers Executed?

Event handlers are executed when a service or host:

* Is in a SOFT problem state
* Initially goes into a HARD problem state
* Initially recovers from a SOFT or HARD problem state.

[SOFT and HARD states are described here](../alerts-notifications/concepts.md#status-types).

### Event Handler Types

There are different types of optional event handlers that you can define to handle host and state changes:

* Global host event handler
* Global service event handler
* Host-specific event handlers
* Service-specific event handlers

> Global host and service event handlers are run for every host or service state change that occurs, immediately prior to
> any host- or service-specific event handler that may be run.

Reviewer: ```suggestion
Global host and service event handlers are executed for every host or service state change that occurs, right before any host- or service-specific event handler is triggered.
```
